src/views/requestAccess/requestAccess.js

```diff
@@ -70,16 +78,41 @@ export const RequestAccess = (props) => {
         Toast(error.message, "error");
         console.log(error);
       });
+    getProperties();
   }, []);
 
+  useEffect(() => {
+    let choices = propertyOptions.filter(
+      p => p.description.toLowerCase().includes(propertySearchText.toLowerCase()))
+    setPropertySearchResults(choices);
+  }, [propertySearchText, propertyOptions])
+  
+  const getProperties = () => {
+    axios.get("/api/properties", makeAuthHeaders(userContext))
+      .then(({ data }) => {
+        let properties = data.properties && data.properties.length > 0
+          ? data.properties.map(property => {
+            return {
+              key: property.id,
+              description: `${property.name}, ${property.address}`
+            }
+          })
+          : data.properties
+        setPropertyOptions(properties);
+        setPropertySearchResults(properties);
+        setShowAddProperty(false);
+      });
+  }
+
   const grantAccess = (role, firstName, lastName, email, id) => {
     // Get role index from roleObject
     let roleID = parseInt(roleObject[role]);
     axios.patch(`/api/user/${id}`, {
       role: roleID,
       firstName: firstName,
       lastName: lastName,
-      email: email
+      email: email,
+      propertyID: propertySelection[0].key
```

Comment:
Probably, this key needs to be propertyID**s** because property managers can be associated with one or multiple properties

`propertyIDs: propertySelection.map(p => p.key)`